pkg/network/additional_networks.go

```diff
@@ -57,3 +58,65 @@ func validateRaw(conf *operv1.AdditionalNetworkDefinition) []error {
 
 	return out
 }
+
+// renderOpenShiftSRIOVNetworkCRD returns the manifests of OpenShiftSRIOV NetworkAttachmentDefinition.
+func renderOpenShiftSRIOVNetworkCRD(manifestDir string) ([]*uns.Unstructured, error) {
+	objs := []*uns.Unstructured{}
+	// render the manifests on disk
+	data := render.MakeRenderData()
+	manifests, err := render.RenderDir(filepath.Join(manifestDir, "network/additional-networks/openshift-sriov/crd"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render OpenShiftSRIOV Network CRD manifests")
+	}
+	objs = append(objs, manifests...)
+	return objs, nil
+}
+
+// renderOpenShiftSRIOVConfig returns the manifests of the SRIOV DaemonSets.
+func renderOpenShiftSRIOVConfig(manifestDir string) ([]*uns.Unstructured, error) {
+	objs := []*uns.Unstructured{}
+	// render the manifests on disk
+	data := render.MakeRenderData()
+	data.Data["SRIOVDevicePluginImage"] = os.Getenv("SRIOV_DEVICE_PLUGIN_IMAGE")
+	manifests, err := render.RenderDir(filepath.Join(manifestDir, "network/additional-networks/openshift-sriov/plugins"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render openshift-sriov additional network manifests")
+	}
+	objs = append(objs, manifests...)
+	return objs, nil
+}
+
+// renderOpenShiftSRIOVCNIConfig returns the OpenShiftSRIOV RawCNIConfig manifests
+func renderOpenShiftSRIOVCNIConfig(conf *operv1.AdditionalNetworkDefinition, manifestDir string) ([]*uns.Unstructured, error) {
+	var err error
+	objs := []*uns.Unstructured{}
+
+	// render SRIOVCNIConfig manifests
+	data := render.MakeRenderData()
+	data.Data["AdditionalNetworkName"] = conf.Name
+	data.Data["AdditionalNetworkConfig"] = conf.SRIOVCNIConfig
+	objs, err = render.RenderDir(filepath.Join(manifestDir, "network/additional-networks/openshift-sriov/cr"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render additional OpenShiftSRIOV network")
+	}
+	return objs, nil
+}
+
+// validateOpenShiftSRIOV checks the AdditionalNetwork config of OpenShiftSRIOV network type.
+func validateOpenShiftSRIOV(conf *operv1.AdditionalNetworkDefinition) []error {
+	out := []error{}
+	var SRIOVConfig map[string]interface{}
+	var err error
+
+	if conf.Name == "" {
+		out = append(out, errors.Errorf("Additional Network Name cannot be nil"))
+	}
+
+	confBytes := []byte(conf.SRIOVCNIConfig)
```

Comment:
Can you add a specific check for an empty CNI config?

Reply:
Done